File: implementations/csharp/tests/Skight.AgentPlatform.MSpec.Tests/MemoryToolSpecs.cs
using System;
using System.Text.Json;
using Machine.Specifications;
using FluentAssertions;
using Skight.AgentPlatform;

namespace Skight.AgentPlatform.MSpec.Tests
{
    [Subject("Memory Tool Module - Persistent Context Store and Recall")]
    public class When_agent_stores_and_recalls_key_value_memory
    {
        Establish context = () =>
        {
            _storeJson = JsonSerializer.Serialize(new { key = "user_preference", value = "Dark Mode" });
            _recallJson = JsonSerializer.Serialize(new { key = "user_preference" });
        };

        Because of = () =>
        {
            _storeResult = MemoryTool.StoreMemoryAsync(_storeJson).GetAwaiter().GetResult();
            _recallResult = MemoryTool.RecallMemoryAsync(_recallJson).GetAwaiter().GetResult();
        };

        It should_confirm_storage_of_memory = () =>
            _storeResult.Should().Contain("user_preference");

        It should_successfully_recall_stored_memory = () =>
            _recallResult.Should().Contain("Dark Mode");

        static string _storeJson;
        static string _recallJson;
        static string _storeResult;
        static string _recallResult;
    }
}

File: implementations/fsharp/tests/Skight.AgentPlatform.FSharp.Tests/MemoryToolSpecs.fs
namespace Skight.AgentPlatform.FSharp.Tests

open System
open System.Text.Json
open Expecto
open Skight.AgentPlatform.FSharp

module MemoryToolSpecs =

    [<Tests>]
    let tests =
        testList "Memory Tool Module Specs" [

            testAsync "Feature: Store and Recall Memory" {
                let storeJson = JsonSerializer.Serialize({| key = "user_pref"; value = "Dark Theme" |})
                let recallJson = JsonSerializer.Serialize({| key = "user_pref" |})

                let! storeRes = MemoryTool.storeMemoryAsync storeJson |> Async.AwaitTask
                let! recallRes = MemoryTool.recallMemoryAsync recallJson |> Async.AwaitTask

                Expect.stringContains storeRes "user_pref" "Store response should mention key"
                Expect.stringContains recallRes "Dark Theme" "Recall response should contain stored value"
            }
        ]
